hwy/base.h

```diff
@@ -1174,14 +1299,14 @@ constexpr bool IsSigned<bfloat16_t>() {
 // Largest/smallest representable integer values.
 template <typename T>
 HWY_API constexpr T LimitsMax() {
-  static_assert(!IsFloat<T>(), "Only for integer types");
-  using TU = MakeUnsigned<T>;
-  return static_cast<T>(IsSigned<T>() ? (static_cast<TU>(~0ull) >> 1)
-                                      : static_cast<TU>(~0ull));
+  static_assert(IsInteger<T>(), "Only for integer types");
+  using TU = UnsignedFromSize<sizeof(T)>;
```

Comment:
Should we change MakeUnsigned to this implementation?

Reply:
The reason why `UnsignedFromSize<sizeof(T)>` is used instead of `MakeSigned<T>` is that it is possible for `!IsSame<size_t, uint32_t>() && !IsSame<size_t, uint64_t>()` to be true on some platforms and the current implementation of `hwy::MakeSigned<T>` will give compiler errors if `!IsIntegerLaneType<T>() && !IsFloat<T>() && !IsSpecialFloat<T>()` is true.

Reply:
I understand why you are using UnsignedFromSize, this makes sense. Let me rephrase: what if we changed the implementation of Make(Un)Signed to what you are doing here, then we could still use Make(Un)Signed here, because it is a bit shorter/more clear?